client/state/data-layer/wpcom/read/site/posts/follow/index.js

```diff
@@ -0,0 +1,85 @@
+/**
```

Comment:
should this be `wpcom/read/sites/posts/follow`? was the missing `s` on `site` intentional?

Reply:
Thanks @dmsnell, you're absolutely right. We current have two other sets of endpoints in the data layer (`/read/site/:site_id/comment_email_subscriptions/*` and `/read/site/:site_id/post_email_subscriptions/*`) which incorrectly use the singular `site` in their endpoint URL. Those endpoints are "unreleased", so I may consider renaming them.

In the meantime, I'll move this endpoint into `wpcom/read/sites/posts/follow` where it belongs 👍